src/pybind/cephfs/cephfs.pyx

```diff
@@ -2165,6 +2165,7 @@ cdef class LibCephFS(object):
             buf = <char *>realloc_chk(buf, _size)
             with nogil:
                 ret = ceph_readlink(self.cluster, _path, buf, _size)
+                buf[ret] = 0
```

Comment:
This access is not safe if `ret < 0` or `ret == size`.

Reply:
That's right, thanks.